### PR TITLE
Feat: Add 'Lock Initiative' button

### PR DESCRIPTION
**Pull request by aMediocreDad**
## Summary
After some consideration, I believe the cleanest way to allow users to use functionality like the 'Veteran'-ability in Dragonbane is to implement it in Year Zero Combat itself (it requires changing how some internal methods execute and we cannot expose that in a configurable manner).

The buttons for locking initiative will now appear by default when 'Reset the Initiative Each Round' is selected.

The resetInitiativeDeck has seen some change in it's functionality as a result. Instead of using 'recall' we now use 'pass' and supply a filtered list of cards this allows us to keep the initiative cards we want to keep between draws in the discard pile.

### PR Type

- [x] This PR is a code change that implements a feature request.
- [ ] This PR adds a new feature that is not an open feature request.
- [x] This PR fixes an issue https://github.com/pafvel/dragonbane/issues/7.
- [ ] This PR is not a code change (e.g. documentation, README, ...)

### Other
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] If code changes were made then they have been tested.
- [x] I have updated the documentation to reflect the changes.

### CODE REVIEW

File: src/utils/utils.js [hunk 1]
```diff
@@ -40,10 +40,14 @@ export function getInitiativeDeckDiscardPile(strict = false) {
  * and shuffles them back into the initiative deck.
  * @param {boolean} [chatNotification=false] Whether to send a chat notification.
```
Comment (Stefouch) [inline src/utils/utils.js:41]: Improving the documentation:
```suggestion
 * @param {boolean} [chatNotification=false] Whether to send a chat notification.
 * @param {number[]} [toExclude=[]] Array of card values to exclude from the reset.
```

**Review by Stefouch — CHANGES_REQUESTED** [review]
Tested and stressed the new settings, and there are lots of issues with this PR:
1. On next round, initiative is reset as expected, but not automatically redrawn.
2. Going back on previous round, previous initiative scores are lost (no memory).
3. Locked initiative does not work for combatant groups (leader is locked, but not other members of the group).

File: src/module/settings.js [hunk 2]
```diff
@@ -73,6 +73,15 @@ export function registerSystemSettings() {
...
 
+  game.settings.register(MODULE_ID, SETTINGS_KEYS.AUTO_SELECT_BEST_CARD, {
+    name: 'SETTINGS.AutoSelectBestCard',
+    hint: 'SETTINGS.AutoSelectBestCardHint',
+    scope: 'world',
+    config: true,
+    type: Boolean,
+    default: false,
+  });
+
```
Comment (aMediocreDad) [inline src/module/settings.js:84]: I realize I have completely misunderstood the intention with the "Automatically Draw Initiative" setting. Only when working with the drawing rules did it dawn on me. 

Seeing that the wording is bad I elected to create a new setting that conveys the intention of the one above and replaces it. A migration could be done, but I think I am not the only one who has misunderstood the intention of said setting (and the effect of not migrating is fairly harmless).

**Review by Stefouch — CHANGES_REQUESTED** [review]
There is a small issue with the lock initiative button: locking initiative before combat start will throw an error when started.
- Would be best to hide the lock button before round 1. ( `game.combat?.round > 0` )
- Also hide the lock initiative button for follower combattants since their init depends on the group leader.

File: src/module/config.js [hunk 3]
```diff
@@ -29,6 +29,16 @@ YZEC.CombatTracker = {
...
+    lockInitiative: [
+      {
+        eventName: 'lock-initiative-button-clicked',
+        label: 'YZEC.CombatTracker.lockInitiative',
+        icon: 'fas fa-lock',
+        id: 'lock-initiative-button',
+        property: 'lockInitiative',
+        visibility: 'owner',
+      },
+    ],
```
[the remark this answers is not in the document]
Comment (aMediocreDad) [reply src/module/config.js:41]: This one is worse. We cannot (should not) serialize conditionals to be evaluated runtime. We could perhaps provide this API through programmatic setup however.

File: src/module/settings.js [hunk 4]
```diff
@@ -73,6 +73,15 @@ export function registerSystemSettings() {
     default: false,
```
Comment (Stefouch) [inline src/module/settings.js:73]: Suggesting `requiresReload = true` so the combat-tracker is refreshed and the lock icon appear/disappear properly.
```suggestion
    default: false,
    requiresReload: true,
```

**Review by Stefouch — APPROVED** [review]
Expect for the single extra suggestion, I think everything is ready for merge.

**Review by Stefouch — APPROVED** [review]

**Pull request merged**